torch_geometric/data/hetero_data.py

```diff
@@ -616,6 +623,50 @@ def _consistent_size(stores: List[BaseStorage]) -> List[str]:
 
         return data
 
+    # :obj:`FeatureStore` interface ###########################################
+
+    def _put_tensor(self, tensor: FeatureTensorType, attr: TensorAttr) -> bool:
+        r"""Stores a feature tensor in node storage."""
+        if not attr.is_set('index'):
+            attr.index = None
+
+        out = self._node_store_dict.get(attr.group_name, None)
+        if out:
+            # Group name exists, handle index:
+            val = getattr(out, attr.attr_name)
+            if val is not None:
+                val[attr.index] = tensor
+        else:
+            # No node storage found, just store tensor in new one:
+            setattr(self[attr.group_name], attr.attr_name, tensor)
+        return True
+
+    def _get_tensor(self, attr: TensorAttr) -> Optional[FeatureTensorType]:
+        r"""Obtains a feature tensor from node storage."""
+        # Retrieve tensor and index accordingly:
+        tensor = getattr(self[attr.group_name], attr.attr_name)
+        if tensor is not None:
```

Comment:
If it requires we set `index` why do we even need to check?

Reply:
Maybe another way to solve this if its confusing is to have a different value for `UNSET` which indicates its going to index 'all'

Reply:
Thanks for the comment! Not sure what you mean by needing to check `index`; this is because `TensorAttr` just requires that its attributes are set (they can be set to None). The current way to index all would be `None`, which I think is acceptable; although I'm happy to define a custom value for the `UNSET` enum to indicate `all` indexing in a follow-up PR.